js/src/ui/Form/Input/input.js

```diff
@@ -111,6 +128,11 @@ export default class Input extends Component {
       textFieldStyle.height = 'initial';
     }
 
+    const underlineStyle = readOnly ? UNDERLINE_READONLY : UNDERLINE_NORMAL;
+    const underlineFocusStyle = focused
+      ? UNDERLINE_FOCUSED
+      : readOnly && typeof focused !== 'boolean' ? { display: 'none' } : null;
```

Comment:
The `{ display: 'none' }` should probably be consistent with the others as a defined constant style.